base/replutil.jl

```diff
@@ -380,6 +380,10 @@ function showerror(io::IO, ex::MethodError)
         end
         print(io, ")")
     end
+    if f_is_function && ( name == :length || name == :size )
```

Comment:
This should check whether the type that was attempted to be called is an iterator (e.g. using `applicable(start, ex.args)`) and iteratorsize is `HasLength`. The error message should then suggest changing the iteratorsize trait or implementing the methods.